lumicks/pylake/kymotracker/detail/trace_line_2d.py

```diff
@@ -180,14 +182,11 @@ def _traverse_line_direction(
     return nodes
 
 
+@dataclass(frozen=True)
 class KymoLineData:
     __slots__ = ["time_idx", "coordinate_idx"]
-
-    def __init__(self, time_idx, coordinate_idx):
-        """Some of the kymotracker classes return numpy arrays. Appending to many small numpy
-        arrays was a bottleneck and therefore we explicitly use lists."""
```

Comment:
We should probably still have a docstring for `KymoLineData` and this comment could be included as I think it's useful. Definitions for `time_idx` and `coordinate_idx` would be good too for consistency in documentation and also to clearly state expected units (pixels)

Reply:
added a proper docstring for it